### DISCUSSION TITLE
Hierarchical data sets

### DISCUSSION THREAD

**Discussion by terryfrankcombe** [mir-group/nequip#234] [General]
Hi folks

Can NequIP be used to generate a single surface that is informed by several levels of theory?  I am thinking here about e.g. sampling widely at some cheap level of theory where we have easy forces, but then doing a subset of configurations at a more accurate, more expensive, higher level of theory.  Can we get a surface that inherits the basic shape from the cheap data, but uses the better data (possibly energy only) to deform that to get a more accurate global surface?

(Of course, one can train on the dense, cheap data, then train on the more sparse, "flatter" difference data and add the two together, but that makes life more complicated.)

Ciao
Terry

**Comment by Linux-cpp-lisp** [comment]
Hi @terryfrankcombe ,

Thanks for your interest in our work!

This is to the best of my knowledge an open question in the literature; I've heard of two approaches suggested but I'm sure you can think of others:
 1. Pre-training on abundant data and then finetuning on expensive higher-level theory data; for info on how to pre-train in our code see this recent discussion: https://github.com/mir-group/nequip/discussions/235
 2. Delta-learning: training two models, one on the abundant data and then other on the presumably simpler difference between the cheap and expensive levels of theory, and then doing final inference using the sum. You can do this in the NequIP framework with some clever pre/post processing of XYZ files, or by making a model builder/module pair that wraps two models, sums their output, and only makes one of them trainable.
